sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisReader.java

```diff
@@ -142,4 +159,30 @@ public Instant getWatermark() {
     return source;
   }
 
+  /**
+   * Returns total size of all records that remain in Kinesis stream after current watermark.
+   * When currently processed record is not further behind than {@link #upToDateThreshold}
+   * then this method returns 0.
+   */
+  @Override
+  public long getTotalBacklogBytes() {
+    Instant watermark = getWatermark();
+    if (watermark.plus(upToDateThreshold).isAfterNow()) {
+      return 0L;
+    }
+    if (backlogBytesLastCheckTime.plus(backlogBytesCheckThreshold).isAfterNow()) {
+      return lastBacklogBytes;
+    }
+    long backlogBytes = lastBacklogBytes;
```

Comment:
Not sure why you need this: why not just do `lastBacklogBytes = kinesis.getBacklogBytes` at line 178, and get rid of the `backlogBytes` variable and line 185?

Reply:
It just seemed more clear probably but it's not needed. I'll apply your suggestion